### PR TITLE
Add ESLint using standard config

### LINKED ISSUE #3: Use shared ESLint config
This project should use the shared ESLint config

### PR DESCRIPTION
**Pull request by Gudahtt**
The standard ESLint config has been applied, with all lint errors fixed. Most of the errors could be fixed automatically.

TypeScript was updated to the latest v4.0 patch because it was a requirement of our ESLint TypeScript config. Also the README example was improved slightly (it now uses `const` instead of `var`, and includes the `assert` import that was missing).

Fixes #3

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@ dist/
 test/*
 !test/index.html
 !test/index.js
+.eslintcache
```
Comment (rekmarks) [inline .gitignore:7]: nit:
```suggestion
eslintcache

```

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      },
+    },
+  ],
+
+  ignorePatterns: [
+    '!.eslintrc.js',
+    '!.prettierrc.js',
+    'dist/',
+    'test/bundle.js',
+    'types',
```
Comment (rekmarks) [inline .eslintrc.js:28]: Could we provide some overrides for this folder instead? IIRC, the rule violations aren't that bad.

**Review by rekmarks — CHANGES_REQUESTED** [review]

**Comment by rekmarks** [comment]
FYI: I committed one suggestion to add a newline at the end of `.gitignore`.

File: .eslintrc.js (hunk 2 above)
Comment (Gudahtt) [reply .eslintrc.js:28]: I don't really understand what ESLint things the type declarations even are. And they're removed in a sibling PR anyway, so it didn't seem important.

**Review by rekmarks — APPROVED** [review]
LGTM!

**Comment by Gudahtt** [comment]
Rebased to resolve conflicts

**Review by rekmarks — APPROVED** [review]

**Pull request merged**